sdk/ahnlich-client-py/ahnlich_client_py/protocol.py

```diff
@@ -1,31 +1,37 @@
 import socket
+from ipaddress import IPv4Address
 
-from ahnlich_client_py import config
+from generic_connection_pool.contrib.socket import TcpSocketConnectionManager
+from generic_connection_pool.threading import ConnectionPool
+
+from ahnlich_client_py.config import service_config
 from ahnlich_client_py.exceptions import AhnlichProtocolException
 from ahnlich_client_py.internals import query, server_response
 
 
 class AhnlichProtocol:
     def __init__(self, address: str, port: int, timeout_sec: float = 5.0):
-        self.address = address
+        self.address = IPv4Address(address)
         self.port = port
-        self.conn = self.connect()
+        self.connection_pool = self.create_connection_pool()
```

Comment:
Instead of using the environmental variables, allow the connection details such as `max_lifetime`, `total_max_size`, e.t.c to be Option[T] and passed into the init and flow into `self.create_connection_pool`, then within create_connection_pool, unwrap them and assign a default, this lets us do something like

```py
client = AhnlichDBClient(
    host="...",
    pool=AhnlichDBPoolSettings(
        max_lifetime_secs=30|None,
        total_max_size=10|None,
     ) | None,
)
```

I think python dataclass can give you that ability to specify defaults and then have people override them

Reply:
Done